test/extended/util/prometheus/helpers.go

```diff
@@ -45,13 +46,50 @@ type prometheusResponseData struct {
 }
 
 // GetBearerTokenURL makes http request with bearer token
-func GetBearerTokenURL(url, bearer string) (string, error) {
-	cmd := fmt.Sprintf("curl --retry 15 --max-time 2 --retry-delay 1 -s -k -H 'Authorization: Bearer %s' %q", bearer, url)
-	output, err := exec.Command("bash", "-e", "-c", cmd).Output()
+func GetBearerTokenURL(url, bearerToken string) (string, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{
+		Timeout:   time.Duration(10 * time.Second),
+		Transport: tr,
+	}
+
+	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		return "", fmt.Errorf("host command failed: %v\n%s", err, output)
+		return "", err
 	}
-	return string(output), nil
+
+	req.Header.Add("Authorization", "Bearer "+bearerToken)
+
+	var resp *http.Response = nil
+
+	ctx := context.Background()
+	condition := func(ctx context.Context) (bool, error) {
+		resp, err = client.Do(req)
+		if err == nil && resp.StatusCode == http.StatusOK {
+			return true, nil
+		}
+
+		if resp != nil {
+			resp.Body.Close()
+		}
+
+		return false, nil
+	}
+	err = wait.PollUntilContextTimeout(ctx, 10*time.Second, 60*time.Second, true, condition)
```

Comment:
I'm not super familiar with golang http conventions but since we create our own context should we not use a `context.WithTimeout(ctx, timeout)` and `PollUntilContextCancel()`?

Reply:
they are equivalent according to the document of `PollUntilContextTimeout` https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollUntilContextTimeout